### ISSUE TITLE
Complete the event directory with two columns relating to the number of cases and contacts involved in the event

### ISSUE THREAD

**Issue by carolinverset** [hzi-braunschweig/SORMAS-Project#3276]
### Situation Description

To date, the event directory presents information characterizing the event, including the number of people participating in the event. It would be relevant to supplement these data with the number of cases and contacts involved in the event.

### Feature Description

In the event directory, create two columns relating to the number of cases on the one hand and the number of contacts on the other. In terms of the number of cases, the only contacts whose source case is also involved in the event will be counted.
This information will also be integrated into the export file (basic and detailed)

### Additional Information
@ftavin @olivierleuci @bernardsilenou

**Comment by Seal33** [comment]
Just so I understand this right, column 1 is supposed to show how many of the event's participants are cases, while column 2 is supposed to tell how many contacts those cases have in total?

**Comment by ftavin** [comment]
Ok with the first column.

The 2nd column should tell how many "contacts" (in general) have their "person" involved in this event.

A checkbox in the filters should be added to add the constraint "source case of the contact is also involved in the event" for the calculation of the 2nd column (that's what @bernardsilenou usually ask for this type of event-contact relationship count)

**Comment by Seal33** [comment]
> Ok with the first column.
> 
> The 2nd column should tell how many "contacts" (in general) have their "person" involved in this event.
> 
> A checkbox in the filters should be added to add the constraint "source case of the contact is also involved in the event" for the calculation of the 2nd column (that's what @bernardsilenou usually ask for this type of event-contact relationship count)

Thank you for your help! I guess each person should only be counted once though, correct? Assuming a scenario where Person A, B and C are part of an event. A and B are cases, and C is a contact of person A and B. A and B are not listed as contacts of one another. In this scenario, would the total number of contacts be 1? Or 2, as C is a contact of both A and B?

**Comment by bernardsilenou** [comment]
@Seal33 @ftavin Thanks for the insight. Here are my suggestions:
1. Since in the other sections of sormas (contact repository, dashboard, etc) we count contacts rather than contact persons, I would recommend we use 2.
2. @Seal33  an extension of your scenario can be that,  resulting case A has a contact with person D and D is not an event participant of the event in question. In this case, D should not contribute to the sum of number of contacts by default. The logic can be:
Number of contact = sum of  unique contacts of all resulting cases condition on the fact that the contact person is in the set of all event participant persons
@ftavin If i can remember well, we decided to implement the default option where all cases and contacts belongs to the event and later add the parameter to include external cases and contact in another issue.

**Comment by ChristopherRiedel** [comment]
@Seal33 
> > Ok with the first column.
> > The 2nd column should tell how many "contacts" (in general) have their "person" involved in this event.
> > A checkbox in the filters should be added to add the constraint "source case of the contact is also involved in the event" for the calculation of the 2nd column (that's what @bernardsilenou usually ask for this type of event-contact relationship count)
> 
> Thank you for your help! I guess each person should only be counted once though, correct? Assuming a scenario where Person A, B and C are part of an event. A and B are cases, and C is a contact of person A and B. A and B are not listed as contacts of one another. In this scenario, would the total number of contacts be 1? Or 2, as C is a contact of both A and B?

Since a contact is not a human but an incident it is mandatory that in this scenario 2 contacts exist.

**Comment by ftavin** [comment]
@bernardsilenou I suggest we implement directly with the parameter to optimize time spent. 2 issues means 2 discussions, two small developments (>1 medium development), 2 PR, 2 reviews...

**Comment by Seal33** [comment]
I implemented the first and second column as discussed. Currently the second column is calculated as follows:

`for each participant (if participant is case ( for all contacts of participant ( if contact is also participant (count + 1))))`

Now for the checkbox, as far as I understand, the checkbox should be listed among the other filters. However it is only supposed to change the way the second column is calculated. 
Checkbox title: "source case of the contact is also involved in the event". If it is ticked we use the calculation method mentioned above. If it is **not** ticked, what exactly am I supposed to calculate? As far as I understood, the new forumla would be

for each contact ( if person is participant in this event (counter + 1))

where _all_ contacts are considered are in the first loop.

Is this correct?

**Comment by Seal33** [comment]
@ftavin @bernardsilenou

**Comment by bernardsilenou** [comment]
@Seal33 Sorry late. The logic it absolute.
I hope we should remember to call this method each time that there in an update in to the entities involved, e.g, more cases are linked linked to event, event, a contact is added or deleted, etc

**Comment by bernardsilenou** [comment]
@ftavin Ok for you too?

**Comment by ftavin** [comment]
@Seal33 @bernardsilenou the logic seems right

**Comment by Seal33** [comment]
> @Seal33 Sorry late. The logic it absolute.
> I hope we should remember to call this method each time that there in an update in to the entities involved, e.g, more cases are linked linked to event, event, a contact is added or deleted, etc

Ok, i currently plan to call the method everytime the list is displayed, so it's always up to date. However I will have to do some research on how much performance this would cost (I don't want to create unnecessary loadingtimes)

**Comment by bernardsilenou** [comment]
@Seal33 I think we may need to see how it function in the production system. This may not be a big deal since we do not load all events at once. @MateStrysewske please can you comment if this is fine?

**Comment by ftavin** [comment]
@Seal33 @bernardsilenou do you think this dev could be integrated in 1.51 ? (PR to be done very quickly)

**Comment by Seal33** [comment]
> @Seal33 @bernardsilenou do you think this dev could be integrated in 1.51 ? (PR to be done very quickly)

I am currently polishing my local build, I think I can commit a PR within the next hour or two. So it would all depend on the speed of the PR

**Comment by Seal33** [comment]
@ftavin Also PR #3351 needs to be merged first.

**Comment by bernardsilenou** [comment]
@ftavin I think we should have it for 1,51 since it would be useful for all the users. But this would all depends on Matés' availability.
@MateStrysewske Any good news from you? Thanks in advance

**Comment by MateStrysewske** [comment]
@MateStrysewske @ftavin We have a feature freeze on the Friday of the week before the release in order to allow the testers to ensure that the features developed for the new version so far are as bug-free as possible. That means that there won't be any more pull requests not related solely to bugfixing merged into the development branch for the 1.51 release, sorry.

**Comment by ftavin** [comment]
@MateStrysewske is this feature freeze systematic (there was one already for 1.50) ? if this is the case, it should be noted somewhere on the milestone... ^^"

**Comment by MateStrysewske** [comment]
Yes, it is - and you're absolutely right, we should add that as a note to the milestones. I wasn't sure how to put it to make sure that developers are aware of it, but that could be a good starting point!

**Comment by Seal33** [comment]
@carolinverset @bernardsilenou Should both columns be sortable?

**Comment by carolinverset** [comment]
Yes please :-)
Best regards

**Comment by AndyBakcsy-she** [comment]
Validated ticket on the latest version on the test.sormas.netzlink.com environment with SORMAS Version: 1.55.0-SNAPSHOT
